File: backend/src/main/java/com/interview/platform/mod04_technology_management/repository/TechnologyRepository.java
package com.interview.platform.mod04_technology_management.repository;

import com.interview.platform.mod04_technology_management.model.TechnologyEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

import java.util.Optional;

public interface TechnologyRepository extends JpaRepository<TechnologyEntity, UUID> {
    Optional<TechnologyEntity> findByName(String name);
}

File: android-app/app/src/main/java/com/interview/platform/domain/repository/TechnologyRepository.kt
package com.interview.platform.domain.repository

import com.interview.platform.domain.model.mod04_technology_management.Technology
import com.interview.platform.data.model.Result
import kotlinx.coroutines.flow.Flow

interface TechnologyRepository {
    fun fetchAll(): Flow<Result<List<Technology>>>
    suspend fun syncData(): Result<Unit>
}
